.github/monitor-typescript-errors/index.js

```diff
@@ -0,0 +1,70 @@
+const fs = require( 'fs' );
+const { getOctokit, context } = require( '@actions/github' );
+const { setFailed, getInput } = require( '@actions/core' );
+const { parseXml, getFilesWithNewErrors } = require( './utils/xml' );
+const { generateMarkdownMessage } = require( './utils/markdown' );
+const { addRecord } = require( './utils/airtable' );
+const { getFileContent } = require( './utils/github' );
+
+const runner = async () => {
+	const token = getInput( 'repo-token', { required: true } );
+	const octokit = getOctokit( token );
+	const payload = context.payload;
+	const repo = payload.repository.name;
+	const owner = payload.repository.owner.login;
+	const fileName = getInput( 'compare', {
+		required: true,
+	} );
+	const newCheckStyleFile = fs.readFileSync( fileName );
+	const newCheckStyleFileParsed = parseXml( newCheckStyleFile );
+	const currentCheckStyleFile = await getFileContent( {
+		octokit,
+		owner,
+		repo,
+		fileName,
+		onFail: setFailed,
+	} );
+
+	if ( ! currentCheckStyleFile.data ) {
+		setFailed( 'No Content Available' );
+		return;
+	}
+
+	const currentCheckStyleFileContentParsed = parseXml(
+		currentCheckStyleFile.data
+	);
+
+	const { header } = generateMarkdownMessage( newCheckStyleFileParsed );
+	const filesWithNewErrors = getFilesWithNewErrors(
+		newCheckStyleFileParsed,
+		currentCheckStyleFileContentParsed
+	);
+
+	const message =
+		header +
+		'\n' +
+		( filesWithNewErrors.length > 0
```

Comment:
If `filesWithNewErrors.length > 0`, should we make the CI job fail to make it more evident? I feel the comment might be easy to miss if the merge button "is green":

![imatge](https://user-images.githubusercontent.com/3616980/198230876-1b522ecd-df33-456e-8043-4ed4fdce9ee2.png)

Reply:
I agree with this strikethrough text from @wavvves's comment https://github.com/woocommerce/woocommerce-blocks/pull/6106#discussion_r1005989244.